nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -699,7 +699,7 @@ def _calculate_taylor_expansion(self, weight_tensor: Tensor, grad: Tensor) -> Te
         return (weight_tensor.detach() * grad.detach()).data.pow(2)
 
     def reset_tools(self):
-        hook_targets = {layer_info.name: layer_info.module.weight for layer_info, _ in self._detect_modules_to_compress()}
+        hook_targets = {name: wrapper.weight for name, wrapper in self.get_modules_wrapper().items()}
```

Comment:
so, weight as a member variable in wrapper always exists? no matter wrapped or unwrapped?

Reply:
only wrapped wrapper has `weight`, the `weight` will be del when unwrapped.

Reply:
Yes, the weight will always exists... Sometimes we need to `_unwrap_model()` to get the model graph, then `_wrap_model()`, so reserving `weight` after unwrapped is necessary.

Reply:
got it

Reply:
I'm just curious, what's the specific benefit of this(allocating the weight and bias as  the member variable) compared to the previous way? Performance or cost less memory?